src/modules/minio/minio.service.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { AccessPolicy, BucketName } from './minio.constants';
+import { isProd } from 'src/shared/helpers';
+import * as Minio from 'minio';
+
+@Injectable()
+export class MinioService {
+  private minioClient: Minio.Client;
+
+  constructor(private readonly configService: ConfigService) {
+    this.minioClient = new Minio.Client({
+      endPoint: this.configService.get('MINIO_HOST'),
+      port: Number(this.configService.get('MINIO_PORT')),
+      useSSL: isProd(this.configService.get('NODE_ENV')),
+      accessKey: this.configService.get('MINIO_ROOT_USER'),
+      secretKey: this.configService.get('MINIO_ROOT_PASSWORD'),
+    });
+  }
+
+  async initializeBucket(
+    bucketName = BucketName.Images,
+    accessPolicy = AccessPolicy.Private,
+  ) {
+    await this.createBucketIfNotExists(bucketName);
+
+    if (accessPolicy == AccessPolicy.Public) {
+      await this.makeBucketPublic(bucketName);
+    }
+  }
+
+  private async makeBucketPublic(bucketName: BucketName) {
+    const bucketPolicyRead = {
+      Version: '2012-10-17',
+      Statement: [
+        {
+          Effect: 'Allow',
+          Principal: '*',
+          Action: ['s3:GetObject'],
+          Resource: [`arn:aws:s3:::${bucketName}/*`],
+        },
+      ],
+    };
+
+    let prevPolicy: typeof bucketPolicyRead;
```

Comment:
It's better to give an initial value to ensure type safety. `... = bucketPolicyRead`;

Reply:
fixed